src/parser_xml.c

```diff
@@ -469,6 +469,14 @@ lydxml_subtree_r(struct lyd_xml_ctx *lydctx, struct lyd_node *parent, struct lyd
         assert(lydctx->parse_opts & LYD_PARSE_OPAQ);
 
         if (xmlctx->ws_only) {
+            /* White string only value should be staticaly allocated.
```

Comment:
Why? The whitespaces may be relevant, the flag is there just for cases when it is not, for a quick check. If you have a `string` value, for example, all the exact whitespace values must be preserved, this change just throws them all away.

Reply:
The white spaces are irrelevant in this case because line 481 changes the value to the empty value anyway. Problems occur when a dynamically allocated string containing only white spaces. If such a string is created then for some reason double free happens. The reason of double free problem is changing the value in some functions to an empty string and don't expecting the value could be dynamically allocated. For example function
 **https://github.com/CESNET/libyang/blob/libyang2/src/tree_data.c#L725-L727**
 **https://github.com/CESNET/libyang/blob/libyang2/src/tree_data.c#L2405-L2407**
 Luckily this cannot happen now in libyang2 because strings with only white characters are statically allocated.
This was a more simple way how to prevent this potential bug.

Reply:
Oh, sorry, I missed the whole context of this code. But I do not like comment, seems more misleading than helpful. If you just put all the code after the `/* ignore WS-only value */`, it should be fine.